# Individual Plan

github link: https://github.com/emily2628/Dsci-100---Individual-Plan.git

## 1. Importing Data

In [1]:
library(tidyverse)
url_players= 'https://drive.google.com/uc?export=download&id=18bYRGtsZOOfM_ZjaapVSedO5kAcqHBNC'
players <- read_csv(url_players)
head(players)

url_sessions= 'https://drive.google.com/uc?export=download&id=1r8jujQRJHYl_B8lU3Mj5ygTgCJHvNrIo'
sessions <- read_csv(url_sessions)
head(sessions)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

## 2. Data Description
The dataset records information about minecraft players and their activity on the research server.
Below are summaries showing the number of unique categories for some key variables, along with basic stats to describe the range of numeric data.
All values are rounded to two decimals where relevant.

### 2.1 Players Table

In [2]:
# distinct values for main categorical variables
distinct(players, experience)
distinct(players, subscribe)
distinct(players, gender)

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

In [3]:
# quick range of numeric variables 
players |> summarise(max(played_hours), min(played_hours), mean = round(mean(played_hours, na.rm = TRUE), 2), 
                     max(Age), min(Age), mean = round(mean(Age, na.rm = TRUE), 2))

max(played_hours) min(played_hours) mean  max(Age) min(Age)
1 223.1             0                 21.14 NA       NA

### Summary of the `Players` Table
There are 196 observations and 7 variables, the variales are:
- `experience` - character; describes the player's experience level (`Beginner`, `Regular`, `Amateur`, `Veteran`, `Pro`,)
- `subscribe` - logical; indicates whether the player subscribed to the game info (`TRUE` or `FALSE`)
- `gender` - character; player's self indentified gender
- `hashed_email` - character; anonymized hashed email address for privacy
- `played_hours` - numeric; total play time in hours (range 0-99)
- `age` - numeric; player's age (range roughly 17-99)
- `player_id` - character; unique identifier for each player

### Potential Issues
1. the variable `experience` is categorical, so it cannot be directly used in numerical analysis. It may need to be encoded later if included in a model.
2. Some gender categories have very few observations compared to other. This could make comparisons between gender groups less reliable.

### 2.2 Sessions Table

In [4]:
# distinct values for main categorical variables
distinct(sessions, hashedEmail)

hashedEmail                                                     
1   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3   f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
5   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
6   96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
7   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
8   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
9   24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
10  88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
11  f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
12  e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
13  c57fa94e436cf49a929d0168e47d26fec3d900b321775e280ef136979c01d5a4
14  6b1cdc07fcc1f7ea09509341fd245dd34fdba386f14a493e9107c18604baedc9
15  a175d4741dc84e6baf77901f6e8e0a06f54809a34e6b5211159bced346f7fb3e
16  18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
17  e3e1bef1ce259c995332635c0ec630a575713451cbe742d7ca93e4c03d0826df
18  4bfad3613c71ace05644bf210195d9fb0d3d9513753ad2f8a16d1a80699bb906
19  42eafe96ed5c1684e3b5cc614d1b01a117173d3ec6898abcf1177d9822bd39f7
20  ed3e6d043dee74fad932be44d0600f4502055198c511927b0c906f50b57d9582
21  8b71f4d66a38389b7528bb38ba6eb71157733df7d1740371852a797ae97d82d1
22  8e0aac3020b3fd9cdef4840b533b4b105aaf1ce1f6f2df3f328ba2a44cecfdc1
23  dc73467f73263dd4a07838330dd1cc115aa3f8b03538911ee6aac048405939c4
24  23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5
25  d13020f8d53c08c95e64f2254424a65c9c479fcfd866b21b850d200a2689a8b5
26  d46bd29a2ed08e3500bd8729085ef4b6f0ca65baf4c756f88b6bedf7a18e0352
27  fb0005032363ec3b30c2ae668f78f5076b7e77de0c6a249e924eee608cd0a1fa
28  d51a8c57269fe347b4b7760ec29c420832cb36a11c47569a9200872831910c05
29  0d4d71be33e2bc7266ee4983002bd930f69d304288a8663529c875f40f1750f3
30  7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e
⋮   ⋮                                                               
96  0d70dd9cac34d646c810b1846fe6a85b9e288a76f5dcab9c1ff1a0e7ca200b3a
97  8e98b6db2053af0bc0e62cd55bcea5a08f23986dec3d02ca520ea039517e6744
98  f7875ae87a61632030d5c4029ee8cf081be7047b2b4a9c04ab5a706f59786197
99  e44041459da2102dc20147ed6f0db4753547be66fc4ddea037bafca64004de2d
100 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
101 119f01b9877fc5ea0073d05602a353b91c4b48e4cf02f42bb8d661b46a34b760
102 d07befbac08be3d876d026ac0458651067001b39081d24724af3c95be76355e0
103 5afb9b4ba7f0bb42d4e4d72ee21be8a51131a7677f0519b4606fdbf770eec750
104 a2a0612e9a7da558cbac2ee3c816740324505a69a6e042c94aa1b67731f9ff01
105 d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db299bd4fedb06a46ad5bb
106 3caa832978e0596779f4ee7c686c4592fb6de893925025d25bf26be9a8cd03f5
107 71e724778a28bb0ebbdacb1328944dfef427525466b012b66e7554a65e401a16
108 ba24bebe588a34ac546f8559850c65bc90cd9d51b821581bd6e25cff437a1081
109 e47ae138603b9a7d6e9b97554bac2daf0477b30e0a4f29ed9d3deacc3c7ec155
110 8eae13e0f5823d9c48fe7b89695ea231f96f644f664bcc74c5a07f7d0a1135bd
111 f9ac013b2f0bc2bd4928a6a0fc8a0aae8b5c4f2670cf9e18f547b7e72034c965
112 2cfed571797b66cc810c32562fc5b0f70b5bec0f5250799245527135bbf9662e
113 c121e4d197469bea90e21c0495001f4e21824adb98cbc6a40d54e66617cee174
114 200db0417ee70863c7e2a0f214c737476fb86de5869d0b907a077d726357e687
115 93efabc9f7de58f44c457bac715cf8f9f44cd88cc574fc02624cac194d2f0ea5
116 7c0ae28a5f85a515a8063f9ed989aa26c5ebcc64f6b7be2948c0b300c2c9bd47
117 4b01bce3f141289709e8278b02ba5d2aaa7105d7ccb9c7deb37670a80e332774
118 bda1905e54b6c745bcced9d59ce655a5bfd03c35cc6abdd8d1610e5c416f79d1
119 e21a324ccf5c873bafe82e47d5137b36aa312ee4803eeb5cbc6cf945f9bb3f38
120 710148466b44cb02ba894aa9005a4892e619250ebd01153055d6c7578fc1ea10
121 fef4e1bed8c3f6dcd7bcd39ab21bd402386155b2ff8c8e53683e1d2793bf1ed1
12

In [5]:
# quick range of numeric variables 
sessions |> 
    summarise(
        latest_start_time = max(start_time, na.rm = TRUE), 
        earliest_start_time = min(start_time, na.rm = TRUE), 
        latest_end_time = max(end_time, na.rm = TRUE), 
        earliest_end_time = min(end_time, na.rm = TRUE),
        latest_original_start = max(original_start_time, na.rm = TRUE), 
        earliest_original_start = min(original_start_time, na.rm = TRUE),
        latest_original_end = max(original_end_time, na.rm = TRUE), 
        earliest_original_end = min(original_end_time, na.rm = TRUE), 
    )

latest_start_time earliest_start_time latest_end_time  earliest_end_time
1 31/08/2024 23:35  01/05/2024 00:58    31/08/2024 23:20 01/05/2024 01:24 
  latest_original_start earliest_original_start latest_original_end
1 1.72733e+12           1.7124e+12              1.72734e+12        
  earliest_original_end
1 1.7124e+12

### Summary of the `Sessions` Table
There are 1535 observations and 5 variables, the variables are:
- `hashedEmail` - anonymized player identifier (used to link to `players` table)
- `start_time` - time when the session began
- `end_time` - time when the session ended
- `original_start_time` - unadjusted start time (before any processing
- `original_end_time` - unadjusted end time (before any processing)

### Potential Issues
1. There are both processed and original timestamps in the dataset. This may indicate that time zone or clock adjustments were made, which could affect how session times are interpreted in later analysis.
2. Some sessions may have very long durations compared to typical gameplay. This could happen if players stayed logged in while inactive or if the server did not record the logout properly, which might influence average playtime results.  

## 3. Questions
### Broad Question
What player characteristics and behaviours are most predictive of subscribing to a game related newsletter, and how do these features differ between various player types 

### Specific Question 
Can player engagement variable such as total playtime (`played_hours`), experience level (`experience`), and number of interactions help predict whether a player subscribes to the game related newsletter (`subscribe`)?

### How the Data Will Help 
The `players` table includes each player's subscription status (`subscribe`) along with their characteristics and activity measures such as `played_hours`, `experience`, and `age`. These variables will help explore whether more engaged or experienced player are more likely to subscribe. 

### To prepare the data, I plan to:
1. Use the `players` table as the main dataset
2. Select `subscribe` as the response variable and variables such as `played_hours`, `experience`, and possibly session based features (from the `sessions` table) as explanatory variables.
3. If needed, join `sessions` and `players` using `hashedEmail` to calculate additional activity metric.
4. Encode categorical variables like `experience` and ensure `subscribe` is represented as a binary outcome.
5. Identify and handle any missing data or outliers that could affect model results. 